### PR TITLE
aws - cloudfront - support fetching with arns

### PR DESCRIPTION
**Pull request by harishachappa**
This PR should provide cloud trail support for CloudFront distribution and Application LoadBalancers. 
With Cloud trail support, we can define below events for active response 

CloudFront:
```
 - source: cloudfront.amazonaws.com
   ids: responseElements.distribution.id
   event: CreateDistribution
 - source: cloudfront.amazonaws.com
   ids: requestParameters.resource
   event: TagResource
```

AppELB:
```
 - source: elasticloadbalancing.amazonaws.com
   ids: responseElements.loadBalancers[0].loadBalancerArn
   event: CreateLoadBalancer
 - source: elasticloadbalancing.amazonaws.com
   ids: requestParameters.resourceArns[0]
   event: AddTags
```

### CODE REVIEW

File: c7n/resources/cloudfront.py [hunk 1]
```diff
@@ -21,6 +21,21 @@ class DescribeDistribution(DescribeSource):
     def augment(self, resources):
         return universal_augment(self.manager, resources)
 
+    def get_resources(self, ids, cache=True):
+        results = []
+        distribution_ids = []
+        for id in ids:
+            # if we get cloudfront distribution arn, we pick distribution id
+            if 'arn:aws:cloudfront' in id:
```
Comment (ajkerrigan) [inline c7n/resources/cloudfront.py:29]: I think this explicit ARN segment will fail outside the standard AWS partition (wouldn't match govcloud or china ARNs). The general flow here seems reasonable, but maybe there's a more flexible option like:
* Checking for `id.startswith('arn:')` (this _seems_ sufficient but I may be oversimplifying)
* Using splits or regexes to catch `arn:<something>:cloudfront:<something>` patterns

**Review by ajkerrigan — CHANGES_REQUESTED** [review]
Thanks for your continued work on WAF support! One comment inline regarding support for non-standard AWS partitions.

File: c7n/resources/appelb.py [hunk 2]
```diff
@@ -80,6 +80,7 @@ class resource_type(TypeInfo):
         arn = "LoadBalancerArn"
         # The suffix varies by type of loadbalancer (app vs net)
         arn_type = 'loadbalancer/app'
+        supports_trailevents = True
```
Comment (kapilt) [inline c7n/resources/appelb.py:83]: this shouldn't be needed, supports_trailevents is default true except for child resources where it is false.

File: c7n/resources/cloudfront.py (hunk 1 above)
Comment (kapilt) [reply c7n/resources/cloudfront.py:29]: we should probably grab the partition from the current region to be partition agnostic, aws gov cloud's don't have cloud front afaik, but china partition does.

File: c7n/resources/appelb.py (hunk 2 above)
Comment (harishachappa) [reply c7n/resources/appelb.py]: Removed this change

File: c7n/resources/cloudfront.py (hunk 1 above)
Comment (harishachappa) [reply c7n/resources/cloudfront.py]: Updated condition to id.startswith('arn:') which should work for Standard AWS partition and China ARNs

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**